### PR TITLE
Allow mdpow analysis scripts to select alchemlyb estimators

### LINKED ISSUE #133: allow mdpow analysis scripts to select alchemlyb estimators
When PR #132 is merged, we can use alchemlyb (including MBAR #26 ) estimators instead of MDPOW TI from the python API. This should also be available from the commandline tools, especially `mdpow-fep`
- [x] commandline options
- [ ] cfg file options

(Decide if one or both.)

### PR DESCRIPTION
**Pull request by VOD555**
Fix #133 
-add options to select alchemlyb estimators in `mdpow-pow`, `mdpow-ghyd` and `mdpow-solvationenergy`
-allow using new estimators with data generated with mdpow of old version

### CODE REVIEW

**Comment by VOD555** [comment]
This should work. I'm going to test it on our workstation.

File: mdpow/fep.py [hunk 1]
```diff
@@ -1323,15 +1337,27 @@ def p_transfer(G1, G2, **kwargs):
...
+        if not hasattr(G, 'start'):
+            G.start = kwargs.pop('start', 0)
+        if not hasattr(G, 'stop'):
+            G.start = kwargs.pop('stop', None)
+        if not hasattr(G, 'SI'):
+            G.SI = kwargs.pop('SI', False)
         if kwargs['force']:
-            G.analyze(**kwargs)
+            if estimator == 'mdpow':
+                G.analyze(**kwargs)
```
Comment (orbeckst) [inline mdpow/fep.py:1349]: Throw an error if someone wants to do mdpwo and not TI
```python
method = kwargs.get('method', 'TI')
if method != "TI":
    errmsg = "Method %s is not implemented in MDPOW, use estimator='alchemlyb'" % method
    logger.error(errmsg)
    raise ValueError(errmsg)
```

File: mdpow/fep.py [hunk 2]
```diff
@@ -1323,15 +1337,27 @@ def p_transfer(G1, G2, **kwargs):
...
+            elif estimator == 'alchemlyb':
+                G.analyze_alchemlyb(**kwargs)
```
Comment (orbeckst) [inline mdpow/fep.py:1351]: Add an `else` here with an error that only 'mdpow' and 'alchemlyb' are available as estimators.

File: mdpow/fep.py [hunk 3]
```diff
@@ -1323,15 +1337,27 @@ def p_transfer(G1, G2, **kwargs):
...
             G.results.DeltaA.Gibbs
             G.logger_DeltaA0()
         except (KeyError, AttributeError):   # KeyError because results is a AttributeDict
             logger.warn("Must analyze simulation because no hydration free energy data available...")
-            G.analyze(**kwargs)
-
+            if estimator == 'mdpow':
+                G.analyze(**kwargs)
+            elif estimator == 'alchemlyb':
+                G.analyze_alchemlyb(**kwargs)
```
Comment (orbeckst) [inline mdpow/fep.py:1360]: Add `else` here ... or better: start at the beginning and check that estimator has the correct value.

File: mdpow/fep.py [hunk 4]
```diff
@@ -1309,11 +1309,25 @@ def p_transfer(G1, G2, **kwargs):
...
+           or ``mdpow`` if you want the default TI method.
+       *method*
+           Use `TI`, `BAR` or `MBAR` method in `alchemlyb`, or `TI` in `mdpow`.
+
     :Returns: (transfer free energy, log10 of the water-octanol partition coefficient = log Pow)
 
     """
 
     kwargs.setdefault('force', False)
+    estimator = kwargs.pop('estimator', 'mdpow')
```
Comment (orbeckst) [inline mdpow/fep.py:1330]: Check that `estimator` is one of the supported ones, then we don't need to check below (and you can ignore my comments below)
```suggestion
    estimator = kwargs.pop('estimator', 'mdpow')
    if not estimator in ('mdpow', 'alchemlyb'):
        errmsg = "estimator = %r is not supported, must be 'mdpow' or 'alchemlyb'" % estimator
        logger.error(errmsg)
        raise ValueError(errmsg)
```

**Review by orbeckst — APPROVED** [review]
Looking good. My suggestions are to add a few error checks. Go ahead!

File: mdpow/fep.py (hunk 1 above)
Comment (orbeckst) [reply mdpow/fep.py:1349]: Not necessary if you check at the top, see code below.

File: mdpow/fep.py (hunk 2 above)
Comment (orbeckst) [reply mdpow/fep.py:1351]: Not necessary if you check at the top, see code below.

File: mdpow/fep.py (hunk 3 above)
Comment (orbeckst) [reply mdpow/fep.py:1360]: Not necessary if you check at the top, see code below.

**Comment by orbeckst** [comment]
@VOD555 get this merged as soon as it works so that @iorga can do his analysis.

**Comment by VOD555** [comment]
@orbeckst I've tried the code locally on our workstation, no problem with mdpow-pow. It can be merged.
It's better to modify `mdpow-pcw` and `mdpow-solvationenergy` in another PR, and I'm going to raise an issue for this.

**Comment by VOD555** [comment]
I'm going to merge the PR.

**Pull request merged**